test/unit/context/test-node-values.js

```diff
@@ -0,0 +1,948 @@
+      sibling2.values.subscribe(Recursive, sibling2Stub);
+      cousin1.values.subscribe(Recursive, cousin1Stub);
+      parent.values.subscribe(Recursive, parentStub);
+      grandparent.values.subscribe(Recursive, grandparentStub);
+      clock.runAll();
+      calcSpy.resetHistory();
+    });
+
+    it('should read a recursive value on the node', async () => {
+      grandparent.values.set(Recursive, 'OWNER1', 'A');
+      clock.runAll();
+
+      expect(grandparentStub).to.be.calledOnce.calledWith('A');
+      expect(parentStub).to.be.calledOnce.calledWith('A');
+      expect(sibling1Stub).to.be.calledOnce.calledWith('A');
+      expect(sibling2Stub).to.be.calledOnce.calledWith('A');
+      expect(cousin1Stub).to.be.calledOnce.calledWith('A');
+      expect(calcSpy).to.have.callCount(5);
+
+      calcSpy.resetHistory();
+      grandparent.values.set(Recursive, 'OWNER1', 'B');
+      clock.runAll();
+
+      expect(grandparentStub).to.be.calledTwice.calledWith('B');
+      expect(parentStub).to.be.calledTwice.calledWith('B');
+      expect(sibling1Stub).to.be.calledTwice.calledWith('B');
+      expect(sibling2Stub).to.be.calledTwice.calledWith('B');
+      expect(cousin1Stub).to.be.calledTwice.calledWith('B');
+      expect(calcSpy).to.have.callCount(5);
+    });
+
+    it('should override a recursive value on the node', async () => {
+      grandparent.values.set(Recursive, 'OWNER1', 'A');
+      clock.runAll();
+
+      expect(grandparentStub).to.be.calledOnce.calledWith('A');
+      expect(parentStub).to.be.calledOnce.calledWith('A');
+      expect(sibling1Stub).to.be.calledOnce.calledWith('A');
+      expect(sibling2Stub).to.be.calledOnce.calledWith('A');
+      expect(cousin1Stub).to.be.calledOnce.calledWith('A');
+      expect(calcSpy).to.have.callCount(5);
+
+      calcSpy.resetHistory();
+      parent.values.set(Recursive, 'OWNER1', 'B');
+      clock.runAll();
+
+      expect(parentStub).to.be.calledTwice.calledWith('B');
+      expect(sibling1Stub).to.be.calledTwice.calledWith('B');
+      expect(sibling2Stub).to.be.calledTwice.calledWith('B');
+      expect(grandparentStub).to.be.calledOnce; // no change.
+      expect(cousin1Stub).to.be.calledOnce; // no change.
+      expect(calcSpy).to.have.callCount(3);
+    });
+
+    it('should recompute a value when a node is removed', async () => {
+      grandparent.values.set(Recursive, 'OWNER1', 'A');
+      parent.values.set(Recursive, 'OWNER1', 'B');
+      clock.runAll();
+
+      expect(grandparentStub).to.be.calledOnce.calledWith('A');
+      expect(parentStub).to.be.calledOnce.calledWith('B');
+      expect(sibling1Stub).to.be.calledOnce.calledWith('B');
+      expect(sibling2Stub).to.be.calledOnce.calledWith('B');
+      expect(cousin1Stub).to.be.calledOnce.calledWith('A');
+      expect(calcSpy).to.have.callCount(5);
+
+      calcSpy.resetHistory();
+      parent.node.remove();
+      await rediscover(parent);
+      clock.runAll();
+      expect(calcSpy).to.have.callCount(0);
```

Comment:
Currently the values are not recomputed on disconnected nodes (a) to avoid extra CPU spent on nodes that will be thrown away, (b) to avoid the wrong state updates since some props would be impossible to compute correctly w/o a proper root.

But there's a test that checks, I think, what you are looking for: "should recompute a value when a node is reparented". It checks that the new parent is used for computes when a node is moved.